### DISCUSSION TITLE
newbie question...can this work with a Python backend?

### DISCUSSION THREAD

**Discussion by bbartling** [bcakmakoglu/vue-flow#565] [Q&A] [answered]
I am tinkering around with some beginner flask and vue tutorials curious if I could use this with Flask back end?

**Comment by bcakmakoglu** [comment, marked as answer]
I don't see why you can't :)
In a professional product I work on, we also use Vue Flow on the frontend to visualize some data processing pipelines and the Backend is written in python.
We just send the output of the `toObject` action by vue flow to the backend, save that somewhere and use that expression again when loading the graph.

How you handle your backend is really up to your preferences 😄

**Reply by bbartling** [reply]
Hi @bcakmakoglu cool stuff....so I have a ton to learn yet on Vue and TypeScript...

So my background is PLC programming for automation industrial technology and the visual flow logic flow tools are sort of what I am used to working with to make stuff Go and I use at the moment Python backend for industrial protocol stacks that I am used to working with.

Hopefully this doesn't sound like a silly question but would it be possible to create super simple logic with Vue Flow: (something like this sequence)

1. 5 AM (adjustable) perform a GET request to a weather API for hourly forecast and parse data
2. Perform if else logic if weather is hot or not in the afternoon
3. If hourly weather forecast is hot logic in the afternoon: DO this to building systems via industrial protocol Python stack stuff
4. else if weather is not hot in the afternoon logic DOES this to building systems via industrial protocol Python stack stuff

I do like Node Red somewhat...but the limitation is having to deal with JavaScript versions of BACnet protocol is a terrible experience from my testing.

Just curious on your incite if something like this is possible or not...it would be super cool if the industrial system logic can be created and then some sort of config file gets past to the Python backend for the logic? Or would it require the browser to open 24/7 for the logic to function? Hopefully this make sense! Thanks for the response!

**Reply by bcakmakoglu** [reply]
I assume what you want to build is similar to this?

![image](https://cdn.discordapp.com/attachments/978041343307841577/1028726047547605073/IMG_0535.gif)

The app in the image was built with vue-flow and shown off on the Discord, so if you're looking to build something similar: Yes sure it's possible, I don't see why it wouldn't :)

**Comment by bbartling** [comment]
Yeah cool...do I need to be a Typescript expert to use VUE flow?

**Reply by bcakmakoglu** [reply]
No, you can use Vue Flow in JS-only as well.
Obviously, I'd rather recommend TypeScript but it's really up to you what you'd rather use :) TS is not a requirement at all.

**Reply by bbartling** [reply]
Okay thanks....the gif you posted is super cool.

Last question...can you tell me how the logic would work with the backend? For example, does the browser need to be open to make the logic work? I am trying to find something that can be setup (hopefully flow logic Vs coding logic) and can be left alone to run on its own with the occasion of dialing into a web dashboard to check on some KPI's or something...hopefully that makes sense, just curious if the browser needs to be open 24/7 to make the logic work. Thanks!

**Reply by bcakmakoglu** [reply]
Well I haven't built the App in the image but the way I think of this is, you create a "pipeline" in the browser and then save it.
The backend handles everything else.

If you configure, on your frontend, a timed execution, like "everyday at 3 PM", your backend needs to be able to handle this case. The frontend should not be concerned with sending execution requests on a time-based manner, otherwise you would have to have your browser or something open all the time, with possibly a socket etc., which would be horrible design of course. Nobody wants to keep their browser open for an application to do it's work at some random time. 😄 

You could also add some manual execution, as seen in the gif above, by pressing a button, which would just send a request to your BE to execute this pipeline immediately.

So the FE should just be concerned with arranging the pipeline in a visual manner, make configurations possible for an end-user etc.
The BE should handle everything concerning the actual business logic that needs to be handled.

**Reply by bbartling** [reply]
Thanks for the quick responses...ill tinker around with this in 2023 : )

This discussion can be closed if "discussions" has this feature. Happy Holidays!

**Reply by bcakmakoglu** [reply]
No problem :)
If you have more questions, feel free to join the Discord and ask as many questions as you need 😄 
Happy holidays to you too!
